src/libraries/System.Utf8String.Experimental/tests/System/Utf8SpanTests.cs

```diff
@@ -348,6 +348,7 @@ public static IEnumerable<object[]> TranscodingTestData()
             yield return new object[] { "Hello" }; // simple ASCII
             yield return new object[] { "a\U00000123b\U00001234c\U00101234d" }; // with multi-byte sequences of varying lengths
             yield return new object[] { "\uF8FF\uE000\U000FFFFF" }; // with scalars from the private use areas
+            yield return new object[] { "\u00E921222324303132333435363738393A3B3C3D3E3F3031323334353637\u00E938393A3B3C3D3E3F" }; // reaches the intrinsics optimizations in System.Text.Unicode.Utf8Utility.GetPointerToFirstInvalidByte
```

Comment:
This test case reaches the code added in this PR.

I copied the value from this other existing test case:
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Runtime/tests/System/Text/Unicode/Utf8UtilityTests.ValidateBytes.cs#L74

It is consumed by a group of unit tests that directly call the internal method `GetPointerToFirstInvalidByte` via reflection.

Reply:
Some more context:
I didn't find any other unit tests reaching my modified code.
I added an exception, right before the `AdvSimd` code I added, with a condition that was always met. Then I ran all the unit tests in `System.Text.Encoding` or `System.Runtime`. I got no hits, aside from the test calling the internal method via reflection.
The internal method is called by `Utf8Span.ToCharArray`, but none of the unit tests verifying this method in `System.Utf8String.Experimental` was reaching my code, so this is why I decided to add that test case.

Reply:
Update: On a closer look, I have to put this test case in a separate unit test. The CI is showing me that some unit tests failed when testing that string. That's because the string is an invalid case, while the rest of the test cases are valid cases.

Reply:
I'm having trouble getting the unit test to hit my modified code in my ARM machine, but as soon as I figure it out I'll submit a commit with that unit test.

Reply:
See https://github.com/dotnet/runtime/pull/38653#issuecomment-659633287 for an update.